lib/active_merchant/billing/gateways/credorax.rb

```diff
@@ -253,10 +254,27 @@ def add_3ds_2_optional_fields(post, options)
 
       private
 
+      def isk_change_check(money)
+        if test?
+          isk_change_time = Time.new(2021, 8, 22, 8, 0, 0, '+00:00')
+        else
+          isk_change_time = Time.new(2021, 8, 31, 8, 0, 0, '+00:00')
+        end
+
+        amount = amount(money)
+
+        if Time.now.utc >= isk_change_time
+          return sprintf('%.0f', amount.to_f / 100)
+        else
+          return amount
+        end
+      end
+
       def add_invoice(post, money, options)
         currency = options[:currency] || currency(money)
 
-        post[:a4] = localized_amount(money, currency)
+        # ISK check can be removed after 2021 August 31 8:01 UTC
+        post[:a4] = currency == 'ISK' ? isk_change_check(money) : localized_amount(money, currency)
```

Comment:
`localized_amount` seems to simply return the `amount` when `non_fractional_currency?`, which the addition of `ISK` to `currencies_without_fractions` seems to trigger.

Could the following work?

```ruby
money = isk_change_check(money) if currency == 'ISK'
post[:a4] = localized_amount(money, currency)
```

Reply:
I'm not sure I understand your concern, but the idea here was that ISK amounts won't get to `localized_amount` until the check is removed, when the updated list will be current. I think your suggestion could also work though, and separates the check even more cleanly. I'll try it out.

Reply:
just wanted to approximate an implementation where we keep the original code, and then just change what needs to change, but both work and once the cut-off date is reached, it won't matter :)

Reply:
Actually I like Brian's simpler suggestion for overriding `non_fractional_currency?` and will probably go with that instead.